src/audio/pipeline/pipeline-schedule.c

```diff
@@ -355,6 +375,49 @@ int pipeline_comp_task_init(struct pipeline *p)
 	return 0;
 }
 
+#if CONFIG_ZEPHYR_DP_SCHEDULER
+static enum task_state dp_task_run(void *data)
+{
+	struct comp_dev *comp = data;
+
+	comp->drv->ops.copy(comp);
+	return SOF_TASK_STATE_RESCHEDULE;
+}
+
+int pipeline_comp_dp_task_init(struct comp_dev *comp)
+{
+	struct task *component_task = NULL;
+	int ret;
+	struct task_ops ops  = {
+		.run		= dp_task_run,
+		.get_deadline	= NULL,
+		.complete	= NULL
+	};
+
+	if (!comp->task) {
+		component_task = rzalloc(SOF_MEM_ZONE_RUNTIME, 0, SOF_MEM_CAPS_RAM,
+					 sizeof(*component_task));
```

Comment:
Design point: there are a lot of independent heap blocks being allocated all at once and then all freed together.  It wouldn't be too hard to block them all together in a struct or something instead.